tests/CMakeLists.txt

```diff
@@ -296,6 +316,13 @@ foreach(target ${test_targets})
     target_compile_options(${target} PRIVATE /utf-8)
   endif()
 
+  if(${CMAKE_CXX_COMPILER_ID} MATCHES "PGI")
+    target_compile_options(${target} PRIVATE -Wc,--pending_instantiations=0)
+    if(NOT WIN32)
+      target_compile_definitions(${target} PRIVATE -D__GCC_ATOMIC_TEST_AND_SET_TRUEVAL=1)
```

Comment:
I think that mysterious define deserves an explanation, same for the --pending_instantiations

Reply:
This is not needed, except for CentOS 7 (and the GCC thing was not needed at all). Added to the CI that runs CentOS 7 and removed from the CMakeLists entirely. :)